src/cursor/session.rs

```diff
@@ -1,44 +1,78 @@
-use std::collections::VecDeque;
+use std::{
+    collections::VecDeque,
+    pin::Pin,
+    task::{Context, Poll},
+};
 
-use futures::future::BoxFuture;
+use futures::{future::BoxFuture, Stream};
+use serde::de::DeserializeOwned;
 
 use super::common::{CursorInformation, GenericCursor, GetMoreProvider, GetMoreProviderResult};
 use crate::{
-    bson::Document,
-    client::ClientSession,
+    bson::{from_document, Document},
     cursor::CursorSpecification,
     error::Result,
     operation::GetMore,
     results::GetMoreResult,
     Client,
+    ClientSession,
     RUNTIME,
 };
 
-/// A cursor that was started with a session and must be iterated using one.
+/// A `SessionCursor` is a cursor that was created with a `ClientSession` and must be iterated using
+/// one. To iterate, retrieve a `SessionCursorHandle` using `SessionCursor::with_session`:
+///
+/// ```rust
+/// # use futures::stream::StreamExt;
+/// # use mongodb::{Client, error::Result, ClientSession, SessionCursor};
+/// #
+/// # async fn do_stuff() -> Result<()> {
+/// # let client = Client::with_uri_str("mongodb://example.com").await?;
+/// # let mut session = client.start_session(None).await?;
+/// # let coll = client.database("foo").collection("bar");
+/// # let mut cursor = coll.find_with_session(None, None, &mut session).await?;
+/// #
+/// while let Some(doc) = cursor.with_session(&mut session).next().await {
+/// println!("{}", doc?)
```

Comment:
Slight nit: could we indent the `println!` here with four extra spaces (beyond the original space after the `///`) so that it renders cleanly in our documentation?